### ISSUE TITLE
Datadog Pagerduty Integration - method not allowed

### ISSUE THREAD

**Issue by taiidani** [terraform-providers/terraform-provider-datadog#160]
### Terraform Version

```
Terraform v0.11.12
+ provider.datadog v1.7.0
+ provider.pagerduty v1.2.1
```

### Affected Resource(s)

```
datadog_integration_pagerduty
```

### Terraform Configuration Files

Setup is part of a larger configuration. Pasting the applicable sections of my submodule here:

```
provider "datadog" {
  version = "~> 1.7"
}

provider "pagerduty" {
  version = "~> 1.2"
}

data "pagerduty_vendor" "datadog" {
  name = "Datadog"
}

resource "pagerduty_service" "app" {
  name                    = "example-for-datadog"
  description             = "Example pagerduty service"
  acknowledgement_timeout = 1800
  auto_resolve_timeout    = "null"
  escalation_policy       = "default"
  alert_creation          = "create_alerts_and_incidents"
}

resource "pagerduty_service_integration" "datadog" {
  name    = "${data.pagerduty_vendor.datadog.name} example-for-datadog"
  service = "${pagerduty_service.app.id}"
  vendor  = "${data.pagerduty_vendor.datadog.id}"
}

resource "datadog_integration_pagerduty" "app" {
  subdomain = "vacasa"

  services = [
    {
      service_name = "${pagerduty_service.app.name}"
      service_key  = "${pagerduty_service_integration.datadog.integration_key}"
    },
  ]
}
```

### Expected Behavior
What should have happened?

A successful update of the Pagerduty integration for Datadog, adding to the existing services on the list.

### Actual Behavior
What actually happened?

```
* datadog_integration_pagerduty.app: Failed to create integration pagerduty using Datadog API: API error 405 Method Not Allowed: {"errors": ["Method Not Allowed."]}
```

### Steps to Reproduce

1. If running the HCL above you may need to tweak the escalation policy and the PD subdomain to something in your installation.
1. `terraform apply`

### Important Factoids

A bit of an off-topic question. It looks like this runs a PUT, but the [API documentation](https://docs.datadoghq.com/api/?lang=curl#pagerduty) indicates that perhaps a POST is better for adding new services? Before running this I had backed up my Datadog integration's configuration as the larger TF configuration is rendering 12+ of these resources over top of the same service configuration, and I had concerns about data loss.

### References

#135 added the functionality that is being tested.

**Comment by bkabrda** [comment]
Yeah, it seems that creating the integration sometimes fails when using the wrong verb, this should be easy to fix. I just think you have POST/PUT switch in your original report - the underlying API client is doing `POST`, but the API docs state that `PUT` should be used.

I'm now working on some PD-related things, so I'll fix this while at it.

**Comment by bkabrda** [comment]
This will be fixed by #237.
